File: practice/algorithms/greedy/flowers/Flowers.scala
import scala.io.Source

object Flowers extends App {
  val lines = Source.stdin.getLines().toList
  val definition = lines(0).split(" ").map(_.toInt)
  val n = definition(0)
  val k = definition(1)
  val prices = lines(1).split(" ").map(_.toInt)
  val descendingPrices = prices.sortWith((i, j) => i > j)
  val total = descendingPrices.indices.map(totalPrice).sum
  println(total)

  def totalPrice(index: Int): Int = {
    val factor = index / k
    (factor + 1) * descendingPrices(index)
  }
}

File: practice/algorithms/greedy/flowers/Flowers.js
'use strict';

const processData = input => {
    let lines = input.split('\n');
    let definition = lines[0].split(' ').map(i => parseInt(i));
    let n = definition[0];
    let k = definition[1];
    let prices = lines[1].split(" ").map(i => parseInt(i));
    let descendingPrices = prices.sort((i, j) => j - i);
    let totalPrice = 0;
    for(let i = 0; i < n; i++) {
        let factor = parseInt(i / k);
        totalPrice += (factor + 1) * descendingPrices[i];
    }
    console.log(totalPrice);

};

process.stdin.resume();
process.stdin.setEncoding("ascii");

var _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));